editor/src/components/inspector/controls/advanced-grid-modal.tsx

```diff
@@ -0,0 +1,283 @@
+  separatorRadixSelectOption,
+} from '../../../uuiui/radix-components'
+import { optionalMap } from '../../../core/shared/optional-utils'
+import type { FlexAlignment } from 'utopia-api/core'
+import { FlexJustifyContent } from 'utopia-api/core'
+
+export interface AdvancedGridModalProps {
+  id: string
+  testId: string
+  key: string
+  popupOpen?: boolean
+  openPopup?: (id: string) => void
+  closePopup?: () => void
+  style?: React.CSSProperties
+  modalOffset?: {
+    x: number
+    y: number
+  }
+}
+
+export const AdvancedGridModal = React.memo((props: AdvancedGridModalProps) => {
+  const modalOffset = props.modalOffset ?? { x: 0, y: 0 }
+  const [dropdownOpen, setDropdownOpen] = React.useState({
+    justifyContent: false,
+    alignContent: false,
+  })
+  const toggleJustifyContentDropdown = React.useCallback((state: boolean) => {
+    setDropdownOpen((prev) => ({ ...prev, justifyContent: state }))
+  }, [])
+  const toggleAlignContentDropdown = React.useCallback((state: boolean) => {
+    setDropdownOpen((prev) => ({ ...prev, alignContent: state }))
+  }, [])
+  const closePopup = React.useCallback(() => {
+    if (!dropdownOpen.justifyContent && !dropdownOpen.alignContent) {
+      props.closePopup?.()
+    }
+  }, [dropdownOpen, props])
+
+  const alignItemsLayoutInfo = useInspectorLayoutInfo('alignItems')
+  const justifyItemsLayoutInfo = useInspectorLayoutInfo('justifyItems')
+  const alignContentLayoutInfo = useInspectorLayoutInfo('alignContent')
+  const justifyContentLayoutInfo = useInspectorLayoutInfo('justifyContent')
+
+  const currentJustifyContentValue = React.useMemo(
+    () => getLayoutInfoValue(justifyContentLayoutInfo),
+    [justifyContentLayoutInfo],
+  )
+
+  const currentAlignContentValue = React.useMemo(
+    () => getLayoutInfoValue(alignContentLayoutInfo),
+    [alignContentLayoutInfo],
+  )
+
+  const currentJustifyItemsValue = React.useMemo(
+    () => getValueOrUnset(justifyItemsLayoutInfo),
+    [justifyItemsLayoutInfo],
+  )
+
+  const currentAlignItemsValue = React.useMemo(
+    () => getValueOrUnset(alignItemsLayoutInfo),
+    [alignItemsLayoutInfo],
+  )
+
+  const contentOptions = [
+    unsetSelectOption,
+    separatorRadixSelectOption(),
+    ...Object.values(FlexJustifyContent).map(selectOption),
+  ]
+
+  const onSubmitJustifyContent = React.useCallback(
+    (value: string) => {
+      if (value === 'unset') {
+        justifyContentLayoutInfo.onUnsetValues?.()
+      } else {
+        justifyContentLayoutInfo.onSubmitValue(value as FlexJustifyContent)
+      }
+    },
+    [justifyContentLayoutInfo],
+  )
+
+  const onSubmitAlignContent = React.useCallback(
+    (value: string) => {
+      if (value === 'unset') {
+        alignContentLayoutInfo.onUnsetValues?.()
+      } else {
+        alignContentLayoutInfo.onSubmitValue(value as FlexJustifyContent)
+      }
+    },
+    [alignContentLayoutInfo],
+  )
+
+  const justifyItemsOptions = React.useMemo(() => itemsOptions('justify'), [])
```

Comment:
I think these two can just be constants, right?

Reply:
yes, moved out of the render function